### DISCUSSION TITLE
Micronaut swagger ui not working with micronaut-security dependencies

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#8907] [General] [answered]
Swagger UI not working when micronaut-security modules added as dependency.

**Steps To Reproduce**
https://github.com/subash89/public-samples/tree/main/hellojaxrs

1. Created a simple app with micronaut jaxrs + swagger. And Swagger UI Working at http://localhost:8080/swagger/views/swagger-ui

2. Now Add micronaut micronaut-security-jwt to the pom.xml and its annotation processor. 

3. Try mvn mn:run and access swagger ui again. **Swagger ui Not showing up**.

Seems some dependency conflict that prevents jaxrs swagger module getting registered. Hence UI not working.

**Comment by graemerocher** [comment, marked as answer]
seems to be working as designed:

```
curl -i http://localhost:8080/swagger/views/swagger-ui
HTTP/1.1 401 Unauthorized
date: Wed, 8 Mar 2023 14:54:28 GMT
connection: keep-alive
transfer-encoding: chunked
```

Micronaut Security is locked down by default so you either need to authentication or open up to allow anonymous access

**Reply by subash89** [reply]
Thanks a lot for checking that. I always used browser to view and didn't realize it's giving 401. 
I had to add all following paths to get this working. 

Can't I make swagger ui work at some custom URL path like http://localhost:8080/openapi ?

   intercept-url-map:
      - pattern: /swagger/views/swagger-ui
        http-method: GET
        access:
          - isAnonymous()
      - pattern: /swagger-ui/res/*
        http-method: GET
        access:
          - isAnonymous()
      - pattern: /swagger/*
        http-method: GET
        access:
          - isAnonymous()
